src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -224,15 +211,15 @@ void ConfigManager::parseConfig(const nlohmann::json& j) {
         j.contains(configurationOptionJsonPosition)) {
       // This will throw an exception, if the json object can't be interpreted
       // with the wanted type.
-      option.setValueWithJson(j.at(configurationOptionJsonPosition));
+      option->setValueWithJson(j.at(configurationOptionJsonPosition));
```

Comment:
I would really like if you would 
use a function that uses
`ranges::transform` on the internal map, s.t. you get `std::pair<const Key&, Option&>`
so
```
for (auto& [key, option] : configurationOptions() {
  option.doSomething(); // not option->
}
```.
In the transform you could the `AD_CORRECTNESS_CHECK` that the unique_ptrs don't point to NULL.
Then you get a really safe interface here, and no pointer syntax in those functions.

Reply:
Done, see [b2e58d4](https://github.com/ad-freiburg/qlever/pull/1040/commits/b2e58d4ca902419ba4b6213529bc4d5f720319f0).